Library/include/CSP/Systems/Assets/AssetSystem.h

```diff
@@ -252,20 +252,31 @@ class CSP_API AssetSystem : public SystemBase
 	CSP_ASYNC_RESULT_WITH_PROGRESS void
 		RegisterAssetToLODChain(const AssetCollection& AssetCollection, const Asset& Asset, int LODLevel, AssetResultCallback Callback);
 
-	// TODO: comment
+	/// @brief Creates a new material backed by an AssetCollection/Asset.
+	/// @param Name const csp::common::String& : The name of the new material.
+	/// @param SpaceId const csp::common::String& : The space id this material is associated with.
+	/// @param Callback GLTFMaterialResultCallback : Callback when asynchronous task finishes.
 	CSP_ASYNC_RESULT void CreateMaterial(const csp::common::String& Name, const csp::common::String& SpaceId, GLTFMaterialResultCallback Callback);
 
-	// TODO: comment
+	/// @brief Updates an existing materials properties.
```

Comment:
[Consider] The documentation here could use a bit more specificity, especially this one, I'm not sure you could figure it out without glancing at source. What happens if the material doesn't exist, and how does the system determine whether the new material you're passing in matches an old one or not?

Reply:
Updated